provider/cmd/pulumi-gen-aws-apigateway/main.go

```diff
@@ -33,24 +34,27 @@ import (
 
 func main() {
 	if len(os.Args) < 4 {
-		fmt.Printf("Usage: %s <language> <out-dir> <schema-file>\n", os.Args[0])
+		fmt.Printf("Usage: %s <language> <version> <out-dir> <schema-file>\n", os.Args[0])
 		os.Exit(1)
 	}
 
-	language, outdir, schemaPath := os.Args[1], os.Args[2], os.Args[3]
+	language, version, outdir, schemaPath := os.Args[1], os.Args[2], os.Args[3], os.Args[4]
 
-	err := emitSDK(language, outdir, schemaPath)
+	err := emitSDK(language, version, outdir, schemaPath)
 	if err != nil {
 		fmt.Printf("Failed: %s", err.Error())
 	}
 }
 
-func emitSDK(language, outdir, schemaPath string) error {
+func emitSDK(language, version, outdir, schemaPath string) error {
```

Comment:
It looks like we could remove `emitSDK` in favor of `pulumi package gen-sdk`. Just an observation.

Reply:
+1
If things need fixing/changing for gen-sdk to be workable let me know. I _really_ want to get people of directly linking to the codegen libraries.

Reply:
We can't currently because the version isn't in the schema and the gen-sdk command doesn't let us override the version on the fly. 

This isn't an issue for other providers because the schema is generated - and the schema for embedding already contains the version, but in this provider we don't generate schema, we've hand written it. We should probably change this at some point, but it's not strictly required for landing this PR.

Reply:
Doesn't `package gen-sdk` have a `--version` flag for this?

Reply:
Oops, oh yeah ... I implemented it 🤦 
- https://github.com/pulumi/pulumi/pull/15970

Reply:
This is now implemented :bushes: